src/ApacheOrcDotNet.OptimizedReader/OrcReader.cs

```diff
@@ -209,14 +207,24 @@ public RowIndex GetColumnRowIndex(int columnId, int stripeId)
                     && s.ColumnId == columnId
                 ).Single();
 
-                var decompressedData = _byteRangeProvider.DecompressByteRange(
-                     rowIndexStream.FileOffset,
-                     rowIndexStream.Length,
-                     _fileTail.PostScript.Compression,
-                     (int)_fileTail.PostScript.CompressionBlockSize
-                 );
+                var compressedBuffer = ArrayPool<byte>.Shared.Rent(rowIndexStream.Length);
+                var decompressedBuffer = ArrayPool<byte>.Shared.Rent(rowIndexStream.Length * 1032);
+                var compressedBufferSpan = compressedBuffer.AsSpan().Slice(0, rowIndexStream.Length);
+                var decompressedBufferSpan = decompressedBuffer.AsSpan().Slice(0, rowIndexStream.Length * 1032);
 
-                return Serializer.Deserialize<RowIndex>(decompressedData.Sequence);
+                try
+                {
+                    _ = _byteRangeProvider.GetRange(compressedBufferSpan, rowIndexStream.FileOffset);
```

Comment:
This is what made me wonder what the return value of `GetRange` was good for.  See notes in HttpByteRangeProvider.